dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -575,28 +577,28 @@ class CZipExtract :
         Progress.SendMessage(PBM_SETRANGE32, 0, gi.number_entry);
         Progress.SendMessage(PBM_SETPOS, 0, 0);
 
-        CStringA BaseDirectory = m_Directory;
-        CStringA Name;
+        CStringW BaseDirectory = m_Directory;
+        CStringW WideName;
         CStringA Password = m_Password;
         unz_file_info64 Info;
         int CurrentFile = 0;
         bool bOverwriteAll = false;
-        while (zipEnum.next(Name, Info))
+        while (zipEnum.next(WideName, Info))
         {
             if (*bCancel)
             {
                 Close();
                 return false;
             }
 
-            bool is_dir = Name.GetLength() > 0 && Name[Name.GetLength()-1] == '/';
+            bool is_dir = WideName.GetLength() > 0 && WideName[WideName.GetLength()-1] == '/';
 
-            char CombinedPath[MAX_PATH * 2] = { 0 };
-            PathCombineA(CombinedPath, BaseDirectory, Name);
-            CStringA FullPath = CombinedPath;
-            FullPath.Replace('/', '\\');    /* SHPathPrepareForWriteA does not handle '/' */
+            WCHAR CombinedPath[MAX_PATH * 2] = { 0 };
+            PathCombineW(CombinedPath, BaseDirectory, WideName);
+            CStringW WideFullPath = CombinedPath;
+            WideFullPath.Replace(L'/', L'\\');    /* SHPathPrepareForWriteA does not handle '/' */
```

Comment:
Is this comment also correct for the W function? Or would that now tolerate the forward slashes? (So question is, whether the line can be stripped or not)

Reply:
Fixed.